android/app/src/main/java/app/organicmaps/car/screens/ErrorScreen.java

```diff
@@ -1,27 +1,80 @@
 package app.organicmaps.car.screens;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.StringRes;
 import androidx.car.app.CarContext;
-import androidx.car.app.model.LongMessageTemplate;
+import androidx.car.app.model.Action;
+import androidx.car.app.model.MessageTemplate;
+import androidx.car.app.model.ParkedOnlyOnClickListener;
 import androidx.car.app.model.Template;
 
 import app.organicmaps.R;
 import app.organicmaps.car.screens.base.BaseScreen;
+import app.organicmaps.car.util.Colors;
 
 public class ErrorScreen extends BaseScreen
 {
-  public ErrorScreen(@NonNull CarContext carContext)
+  @StringRes
+  private final int mErrorMessage;
+
+  private final boolean mIsCloseable;
+
+  private ErrorScreen(@NonNull Builder builder)
   {
-    super(carContext);
+    super(builder.mCarContext);
+    mErrorMessage = builder.mErrorMessage;
+    mIsCloseable = builder.mIsCloseable;
   }
 
   @NonNull
   @Override
   public Template onGetTemplate()
   {
-    final LongMessageTemplate.Builder builder = new LongMessageTemplate.Builder(getCarContext().getString(R.string.dialog_error_storage_message));
-    builder.setTitle(getCarContext().getString(R.string.dialog_error_storage_title));
+    final MessageTemplate.Builder builder = new MessageTemplate.Builder(getCarContext().getString(mErrorMessage));
+    builder.setHeaderAction(Action.APP_ICON);
+    builder.setTitle(getCarContext().getString(R.string.app_name));
+    if (mIsCloseable)
+    {
+      builder.addAction(new Action.Builder()
+          .setBackgroundColor(Colors.BUTTON_ACCEPT)
+          .setTitle(getCarContext().getString(R.string.close))
+          .setOnClickListener(ParkedOnlyOnClickListener.create(this::finish)).build()
+      );
+    }
 
     return builder.build();
   }
+
+  public static class Builder
```

Comment:
Why do we need this builder at all? What is wrong with `new MessageTemplate.Builder(getCarContext())` builder?

Reply:
MessageTemplate is used for templates, not screens. For screens that accept 3+ arguments in the constructor, I prefer to use the Builder pattern, as demonstrated in the CarApp library.

Reply:
ok, let's revise all builders, utils classes and other abstractions after getting all features implemented. It is not a blocker for now.